### ISSUE TITLE
KeyError: 'refdoc'

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#639]
Hi

In the terminal I get:

**Exception occurred:

  **File "/usr/lib/pymodules/python2.6/docutils/nodes.py", line 537, in **getitem**

```
**return self.attributes[key]
```

**KeyError: 'refdoc'

**The full traceback has been saved in /tmp/sphinx-err-nHHrpl.log, if you want to report the issue to the developers.**

The problem occurs when you refer to a non-existent reference in your code documentation. 

And the error is actually in docutils instead, I guess.

**Workaround:

If you change the code at line 537 in nodes.py so you get a print of the self.attributes in the terminal then you can see the reference that causes the problem.

Regards/brownie

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/639
- Originally reported by: Niels Henrik Bruun
- Originally created at: 2011-02-16T13:02:36.910

**Comment by shimizukawa** [comment]
_From Robert Lehmann on 2011-02-16 12:47:23+00:00_

During which operation did the error occur?  Please attach the referenced log and, if possible, a minimal reproducing source file.  (The output you produced with your workaround could be helpful too.)  Do you observe the same behaviour with the latest Sphinx (in case you're running a legacy version)?

**Comment by shimizukawa** [comment]
_From Anonymous on 2011-02-16 16:11:46+00:00_

hi
My sphinx source is rather big so I'll prefer to explain when it happens.
It's when you do a make html from scratch, ie _build is empty

The whole terminal log is shown below.

I attach the log file.

Regards/nhb

nhb@Linux-netbook:~$ cd '/home/nhb/PythonHacks'

nhb@Linux-netbook:~/PythonHacks$ make html

sphinx-build -b html -d _build/doctrees   . _build/html

Making output directory...

Running Sphinx v0.6.4

loading pickled environment... not found

building [html]: targets for 15 source files that are out of date

updating environment: 15 added, 0 changed, 0 removed

reading sources... [ 53%] rstFiles/300_Thoughts_on_finance/301 On Time Value and
reading sources... [ 86%] rstFiles/500_Notes_on_spreadsheets_etc/An example of M
reading sources... [100%] rstFiles/999 aboutMe  s
rstFiles/400_Mathematical_notes/latex_math.rst:34: (ERROR/3) Unexpected indentation.

/home/nhb/PythonHacks/rstFiles/400_Mathematical_notes/SplineAtScipy.rst:: WARNING: image file not readable: rstFiles/400_Mathematical_notes/400_Mathematical_notes
/Graphics/naturalCubicSplineExample.png

/home/nhb/PythonHacks/rstFiles/400_Mathematical_notes/SplineAtScipy.rst:: WARNING: image file not readable: rstFiles/400_Mathematical_notes/400_Mathematical_notes/Graphics/compareNaturalSplineAndScipySpline.png

/home/nhb/PythonHacks/rstFiles/400_Mathematical_notes/latex_math.rst:34: (ERROR/3) Unexpected indentation.

/home/nhb/PythonHacks/rstFiles/500 Notes on spreadsheets etc.rst:18: (WARNING/2) Block quote ends without a blank line; unexpected unindent.

/home/nhb/PythonHacks/rstFiles/500_Notes_on_spreadsheets_etc/An example of Method and operator overload in spreadsheets.rst:8: (WARNING/2) Block quote ends without a blank line; unexpected unindent.

/home/nhb/PythonHacks/rstFiles/500_Notes_on_spreadsheets_etc/An example of Method and operator overload in spreadsheets.rst:: WARNING: image file not readable: rstFiles/500_Notes_on_spreadsheets_etc/500_Notes_on_spreadsheets_etc/pictures/MethodAndOperatorOverload_GNumeric.png

looking for now-outdated files... none found

pickling environment... done

checking consistency... /home/nhb/PythonHacks/_static/PythonHacks/financeRoot/Licence.rst:: WARNING: document isn't included in any toctree

/home/nhb/PythonHacks/_static/PythonHacks/financeRoot/README.rst:: WARNING: document isn't included in any toctree

/home/nhb/PythonHacks/rstFiles/300_Thoughts_on_finance/301 On Time Value and dateflows.rst:: WARNING: document isn't included in any toctree

/home/nhb/PythonHacks/rstFiles/400_Mathematical_notes/SplineAtScipy.rst:: WARNING: document isn't included in any toctree

/home/nhb/PythonHacks/rstFiles/400_Mathematical_notes/latex_math.rst:: WARNING: document isn't included in any toctree

/home/nhb/PythonHacks/rstFiles/500_Notes_on_spreadsheets_etc/An example of Method and operator overload in spreadsheets.rst:: WARNING: document isn't included in any toctree

done

preparing documents... done

writing output... [ 40%] rstFiles/200 PythonHacksnceRoot/READMEe

**Exception occurred:

  **File "/usr/lib/pymodules/python2.6/docutils/nodes.py", line 537, in **getitem**

```
**return self.attributes[key]
```

**KeyError: 'refdoc'

**The full traceback has been saved in /tmp/sphinx-err-yOtpZj.log, if you want to report the issue to the developers.

Please also report this if it was a user error, so that a better error message can be provided next time.

Either send bugs to the mailing list at http://groups.google.com/group/sphinx-dev/, or report them in the tracker at https://bitbucket.org/birkenfeld/sphinx/issues/. Thanks!

make: **\* [html] Fejl 1

nhb@Linux-netbook:~/PythonHacks$

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-09-22 12:18:51+00:00_

This is already fixed in 1.0.x.

**Comment by riebling** [comment]
Seems to have been fixed for a while now. I know nothing about sphinx, just using someone else's system that used to work. But I sure get a similar looking (and similarly perplexing) error message - here's the log:
```
# Sphinx version: 1.3.1
# Python version: 2.7.15 (CPython)
# Docutils version: 0.12 release
# Jinja2 version: 2.7.3
# Last messages:
#   none found
#   pickling environment...
#   done
#   checking consistency...
#   done
#   preparing documents...
#   done
#   writing output... [  8%] WCEtool
#   writing output... [ 16%] extra-tools
#   writing output... [ 25%] formats
# Loaded extensions:
#   alabaster (0.7.3) from /home/vagrant/anaconda/lib/python2.7/site-packages/alabaster/__init__.pyc
Traceback (most recent call last):
  File "/home/vagrant/anaconda/lib/python2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/cmdline.py", line 245, in main
    app.build(opts.force_all, filenames)
  File "/home/vagrant/anaconda/lib/python2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/application.py", line 264, in build
    self.builder.build_update()
  File "/home/vagrant/anaconda/lib/python2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/builders/__init__.py", line 245, in build_update
    'out of date' % len(to_build))
  File "/home/vagrant/anaconda/lib/python2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/builders/__init__.py", line 316, in build
    self.write(docnames, list(updated_docnames), method)
  File "/home/vagrant/anaconda/lib/python2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/builders/__init__.py", line 354, in write
    self._write_serial(sorted(docnames), warnings)
  File "/home/vagrant/anaconda/lib/python2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/builders/__init__.py", line 360, in _write_serial
    doctree = self.env.get_and_resolve_doctree(docname, self)
  File "/home/vagrant/anaconda/lib/python2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/environment.py", line 1291, in get_and_resolve_doctree
    self.resolve_references(doctree, docname, builder)
  File "/home/vagrant/anaconda/lib/python2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/environment.py", line 1549, in resolve_references
    newnode = self._resolve_any_reference(builder, node, contnode)
  File "/home/vagrant/anaconda/lib/python2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/environment.py", line 1636, in _resolve_any_reference
    refdoc = node['refdoc']
  File "/home/vagrant/anaconda/lib/python2.7/site-packages/docutils/nodes.py", line 567, in __getitem__
    return self.attributes[key]
KeyError: 'refdoc'
```
